kratos/utilities/discont_utils.h

```diff
@@ -519,255 +519,236 @@ class DiscontinuousShapeFunctionsUtilities
+        unsigned int partition_number = 0;
+        for (unsigned int i=0; i<4; i++) 
         {
-            unsigned int j_aux = i + 2;
-            if (j_aux>2) j_aux -= 3;
-            boost::numeric::ublas::bounded_matrix<int,3,2> partition_father_nodes;
-            array_1d<double,3> N;
+            array_1d<double, 3> N;
+            bounded_matrix<int, 3, 2> partition_father_nodes;
+
+            const unsigned int j_aux = (i+2 > 2) ? i - 1 : i + 2;
+
             if (i<3)
             {
-                partition_father_nodes(0,0)=i;
-                partition_father_nodes(0,1)=i;
-                partition_father_nodes(1,0)=aux_nodes_father_nodes(i,0); //we are using i aux node
-                partition_father_nodes(1,1)=aux_nodes_father_nodes(i,1); //we are using i aux node
-                partition_father_nodes(2,0)=aux_nodes_father_nodes(j_aux,0); //we are using j_aux node
-                partition_father_nodes(2,1)=aux_nodes_father_nodes(j_aux,1); //we are using j_aux node
-
-                coord_subdomain(0,0)=rPoints(i,0);
-                coord_subdomain(0,1)=rPoints(i,1);
-                coord_subdomain(1,0)=aux_points(i,0);
-                coord_subdomain(1,1)=aux_points(i,1);
-                coord_subdomain(2,0)=aux_points(j_aux,0);
-                coord_subdomain(2,1)=aux_points(j_aux,1);
+                partition_father_nodes(0,0) = i;
+                partition_father_nodes(0,1) = i;
+                partition_father_nodes(1,0) = aux_nodes_father_nodes(i,0); //we are using i aux node
+                partition_father_nodes(1,1) = aux_nodes_father_nodes(i,1); //we are using i aux node
+                partition_father_nodes(2,0) = aux_nodes_father_nodes(j_aux,0); //we are using j_aux node
+                partition_father_nodes(2,1) = aux_nodes_father_nodes(j_aux,1); //we are using j_aux node
+
+                coord_subdomain(0,0) = rPoints(i,0);
+                coord_subdomain(0,1) = rPoints(i,1);
+                coord_subdomain(1,0) = aux_points(i,0);
+                coord_subdomain(1,1) = aux_points(i,1);
+                coord_subdomain(2,0) = aux_points(j_aux,0);
+                coord_subdomain(2,1) = aux_points(j_aux,1);
             }
-            else
+            else // Last partition, composed by the 3 aux nodes
             {
-                //the last partition, made by the 3 aux nodes.
-                partition_father_nodes=aux_nodes_father_nodes;
-                coord_subdomain=aux_points;
+                coord_subdomain = aux_points;
+                partition_father_nodes = aux_nodes_father_nodes;
             }
-            //calculate data of this partition
+
+            // Calculate this partition data
             double temp_area;
             CalculateGeometryData(coord_subdomain, DN_DX_subdomain, temp_area);
-            if (temp_area > 1.0e-20) //ok, it does not have zero area
+
+            if (temp_area > 1.0e-20) // Check if the partition has zero area
```

Comment:
I recommend to use epsilon http://en.cppreference.com/w/cpp/types/numeric_limits/epsilon